.github/workflows/lint.yml

```diff
@@ -1,53 +1,76 @@
 name: Lint
 
-on: [push]
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - stable*
 
-jobs:
   php-linters:
     runs-on: ubuntu-latest
+
     strategy:
       matrix:
         php-versions: ['7.2', '7.3', '7.4']
+
     name: php${{ matrix.php-versions }} lint
     steps:
       - name: Checkout
         uses: actions/checkout@master
+
       - name: Set up php${{ matrix.php-versions }}
         uses: shivammathur/setup-php@master
         with:
           php-version: ${{ matrix.php-versions }}
           coverage: none
+
       - name: Lint
         run: "find . -name \\*.php -not -path './vendor/*' -exec php -l \"{}\" \\;"
 
   style-linter:
     runs-on: ubuntu-latest
+
     name: Style lint
+
     steps:
       - name: Checkout
         uses: actions/checkout@master
+
       - name: Set up Node
         uses: actions/setup-node@v1
         with:
           node-version: 12.x
+
       - name: Set up Stylelint 11
         run: npm i -g stylelint@11.0.0
+
       - name: Style lint
         run: $(which stylelint) -f verbose css
 
   eslint:
     runs-on: ubuntu-latest
-    name: ESLint
+
+    strategy:
+      matrix:
+        node-versions: [12.x]
+
+    name: node${{ matrix.node-versions }}
+
     steps:
       - name: Checkout
         uses: actions/checkout@master
-      - name: Set up Node
+
+      - name: Set up node ${{ matrix.node-versions }}
         uses: actions/setup-node@v1
         with:
-          node-version: 12.x
+          node-version: ${{ matrix.node-versions }}
+
       - name: NPM install
```

Comment:
Names differs from the original file
https://github.com/nextcloud/app-tutorial/blob/master/.github/workflows/lint.yml